### PR TITLE
Fix caching related issues

### PR DESCRIPTION
**Pull request by mistercrunch**
Moving to cache the output of the dataframe as opposed to the output of `get_data`. Caching the output of `get_data` doesn't work if the basis for the `cache_key` is the "query object". When changing controls like the moving average configuration, this doesn't affect the query object (same cache key), but it does affect the post processing that append in `get_data`. If we use the query object as the basis for the cache key, we need to cache the dataframe itself, and re-execute get_data against the cache.

This may not be 100% ready for merging as:
* I'm unclear on whether the serialization of the dataframe (using df.to_dict) is 100% clean, there could be bugs where some things get lost through the serde process
* FilterBox doesn't get cached anymore as it doesn't follow the common pattern of using `get_df

@john-bodley 

Also the introduction of a `rendered` chartState seemed to have broken the `cache` tag and the row label in the UI.

### CODE REVIEW

**Comment by mistercrunch** [comment]
It seems like we may want to re-think/cleanup the caching logic a bit.

**Comment by mistercrunch** [comment]
For the record, I'm doing more work on this. Moving to pickling dataframes as other methods of serializations seem incomplete/risky.

**Comment by graceguo-supercat** [comment]
@mistercrunch Hi Max, do you mind if i create a another PR just to fix JS part? So this PR focus on cache issue?

**Comment by mistercrunch** [comment]
@john-bodley I think I've got it mostly nailed down now at this point. This turned out to be a much more complex headache then I thought originally. Especially in the area of caching properly multi-queries visualizations.

File: superset/assets/package.json [hunk 1]
```diff
@@ -93,8 +93,8 @@
     "react-sortable-hoc": "^0.6.7",
     "react-split-pane": "^0.1.66",
     "react-syntax-highlighter": "^5.7.0",
-    "react-virtualized": "^9.3.0",
-    "react-virtualized-select": "^2.4.0",
+    "react-virtualized": "9.3.0",
```
Comment (mistercrunch) [inline superset/assets/package.json]: Unrelated but that fixed the build

**Comment by john-bodley** [comment]
@mistercrunch overall this LGTM. We'll probably have to test it extensively in our staging environment.

File: superset/viz.py [hunk 2]
```diff
@@ -67,6 +67,40 @@ def __init__(self, datasource, form_data):
...
+
+        # Keeping track of whether some data came from cache
+        # this is useful to trigerr the <CachedLabel /> when
+        # in the cases where visualization have many queries
+        # (FilterBox for instance)
+        self._some_from_cache = False
+        self._any_cache_key = None
+        self._any_cached_dttm = None
+
+        self.run_extra_queries()
```
[the remark this answers is not in the document]
Comment (mistercrunch) [reply superset/viz.py]: Oh wow right. This should not be here. Let me take a shot at it.

[the remark this answers is not in the document]
Comment (mistercrunch) [reply superset/viz.py]: This should do it: https://github.com/apache/incubator-superset/pull/4627 , mind trying it on you side?

[the remark this answers is not in the document]
Comment (michellethomas) [reply superset/viz.py]: thanks for the quick fix! I'll test

**Pull request merged**